### PR TITLE
Add cross build for Scala 3

### PR DESCRIPTION
**Pull request by ybasket**
Changes in detail:
* Add Scala 3.0.0 to build
* Remove superfluous silencer plugin (not needed on 2.12+)
* Adjust to new sbt syntax
* Rewrite `Symbol` syntax to be Scala 3 compatible without special import

### CODE REVIEW

**Comment by blemale** [comment]
Hi @ybasket ! Thanks a lot for this contribution.
It seems the CI is failing due to scoverage, see [here](https://github.com/blemale/scaffeine/pull/211/checks?check_run_id=2930283486#step:5:255). I will try to take a look later in the week.

**Comment by ybasket** [comment]
> Hi @ybasket ! Thanks a lot for this contribution.
> It seems the CI is failing due to scoverage, see [here](https://github.com/blemale/scaffeine/pull/211/checks?check_run_id=2930283486#step:5:255). I will try to take a look later in the week.

Hi @blemale, you're welcome. I just pushed an attempt of fixing the coverage problem on CI – the pipeline needs your approval, so I can't see yet whether I got it right, but should be close. It runs a job per Scala version which has the drawback that they have to be specified redundantly (in CI and in `build.sbt`), but the advantage (despite fixing Scala 3) that CI pipelines should run faster as the different Scala versions are checked in parallel and not sequentially.

**Review by blemale — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -3,13 +3,18 @@ on: [push,pull_request]
...
+        scala: ["2.12.14", "2.13.6", "3.0.1"]
     steps:
       - uses: actions/checkout@v2
       - uses: coursier/cache-action@v6
       - uses: olafurpg/setup-scala@v11
         with:
           java-version: zulu@1.11
-      - run: sbt clean scalafmtSbtCheck scalafmtCheck coverage +test +doc
-      - run: sbt coverageReport coveralls
+      - run: sbt ++{{ matrix.scala }} clean scalafmtSbtCheck scalafmtCheck test doc
```
Comment (blemale) [inline .github/workflows/ci.yml:15]:
```suggestion
      - run: sbt ++${{ matrix.scala }} clean scalafmtSbtCheck scalafmtCheck test doc
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -3,13 +3,18 @@ on: [push,pull_request]
...
+        if: startsWith(matrix.scala, '3')
+      - run: sbt ++{{ matrix.scala }} clean scalafmtSbtCheck scalafmtCheck coverage test doc coverageReport coveralls
```
Comment (blemale) [inline .github/workflows/ci.yml:17]:
```suggestion
      - run: sbt ++${{ matrix.scala }} clean scalafmtSbtCheck scalafmtCheck coverage test doc coverageReport coveralls
```

**Comment by blemale** [comment]
@ybasket I will release a new version with a scala 3 artifact as soon as I have access to my laptop, ie next week. Thanks again for your contribution !

**Comment by blemale** [comment]
@ybasket I have released a 5.1.0 version with a scala 3 artifact, see [here](https://search.maven.org/artifact/com.github.blemale/scaffeine_3/5.1.0/jar).

**Comment by ybasket** [comment]
Merci beaucoup!

**Comment by catostrophe** [comment]
Can we have the same for 4.x?

**Pull request merged**